### ISSUE TITLE
Highlight H2 push in the waterfall

### ISSUE THREAD

**Issue by soulgalore** [micmro/PerfCascade#198]
@tobli has an idea of how we can show pushes directly inside the waterfall graph. @worenga provided us a couple of H2 push URLs that we can use for testing:

> https://www.nghttp2.org https://www.filamentgroup.com https://www.middagsfrid.se https://www.forrent.com https://www.mnot.net https://www.benediktwolters.de https://www.eaglesecurity.com https://yeswebsites.net https://www.travelground.com

**Comment by soulgalore** [comment]
See https://github.com/micmro/PerfCascade/issues/119#issuecomment-273044477:

> ![push](https://cloud.githubusercontent.com/assets/540266/22011927/cc10ef46-dc91-11e6-9443-2575bf850331.jpg)
> 
> This is roughly my idea for moving the push arrow closer to where the server actually issues the push. Saves space to the left, and maybe makes slightly more semantical sense. The arrow is slightly lighter as well.

**Comment by micmro** [comment]
Options to check if a response was pushed:
- `_was_pushed` (WPT, browsertime) - already added to the left legend by @tobli 
- `x-http2-push` header (added by e.g. h2o, [nghttp2](https://github.com/nghttp2/nghttp2/issues/352), caddy)

Perhaps we can add some more info about the push when hovering over the icon (looks like WPT only)?
```
...
   "_http2_stream_id": 39,
   "_http2_stream_dependency": 11,
   "_http2_stream_weight": 2,
   "_http2_stream_exclusive": 0,
...
```

For future reference: Another good source for http2 sites would be [this httpArchive BigQuery Script](https://discuss.httparchive.org/t/which-origins-use-http-2-server-push/795)



[WPT and browsertime HARs for filamentgroup.com](https://github.com/micmro/PerfCascade/files/1107485/HARs_filamentgroup.com.zip)

**Comment by soulgalore** [comment]
Yep we should pickup the header too, good idea. I think @tobli added a screenshot somewhere on how we can highlight it in the waterfall.

**Comment by micmro** [comment]
Added the screenshot and @tobli's comment as quote to your initial reference.

**Comment by worenga** [comment]
The x-http2-push header is in no shape or Form standard. E.g. CDNs such as Cloudflare are using different ones such as cf-h2-pushed. 

Instead of incorporating this zoo of possibilities we should concentrate on Browser feedback

**Comment by micmro** [comment]
I'm aware it's not a standard, but better than nothing.

What exactly do you mean with "Browser feedback"?

**Comment by soulgalore** [comment]
Yes think we should add those extra WPT only, we haven't had the chance to add it browsertime but when we do, we can follow the same standard.

I think it good for us to catch what we can and then when HAR spec 2.0 comes out we can follow the new standard :D

**Comment by micmro** [comment]
Still working on it, but here's a quick preview:
<img width="656" alt="screen shot 2017-06-30 at 3 36 49 pm" src="https://user-images.githubusercontent.com/1680390/27723851-fc6806f4-5da9-11e7-8ad5-408f6666b33c.png">

Looks a bit funny with no block size though.
Any thoughts?

**Comment by tobli** [comment]
I like it (but I'm biased ;-). Good work!
Are you thinking this should be the default, or available as an option? Having push arrows both inline and on the left looks a bit strange to me.

**Comment by soulgalore** [comment]
@micmro looks great! I agree with @tobli we should disable the ones to the left when we got the ones inline.

**Comment by micmro** [comment]
Thanks. I meant to da that, but have not gotten around to do so :).
@tobli I'd say we keep the inline ones default and make the ones on the left ones switchable: `show`, `showBoth`, `hide` (default) or so.
I also want to play a bit with the colour to make it a bit more subtle, and perhaps use the font's arrows.

**Comment by micmro** [comment]
Finally (after a very long time) I got around to work on this issue again.
Replaced the arrow with font arrows and removed them form the icon list, rather than making in configurable. Just the hover info does not show up unfortunately.

You can checkout [branch `198`](https://github.com/micmro/PerfCascade/tree/198) for the code changes.

<img width="741" alt="screen shot 2018-02-24 at 12 06 49 am" src="https://user-images.githubusercontent.com/1680390/36621137-d7a3b08a-18f6-11e8-8496-e342e1daef2c.png">

**Comment by micmro** [comment]
Released via `v2.5.0`
